File: testresources/java/accuracy/src/domain/indirect/violatingfrom/CallInstanceMethodIndirectIndirect_MethodVarMethod.java
package domain.indirect.violatingfrom;

import domain.indirect.intermediate.BackgroundService;

public class CallInstanceMethodIndirectIndirect_MethodVarMethod {
	private BackgroundService bckg;
	private String test;
	
	public CallInstanceMethodIndirectIndirect_MethodVarMethod() {
		bckg = new BackgroundService();
	}

	// Double indirect access via normal method 
	public String case6(){
			test = (String) bckg.getServiceTwo().serviceOne.getName();
		return test;
	}
}

File: src/test/resources/csharp/benchmark/Domain/Indirect/AllowedFrom/CallInstanceMethodIndirectIndirect_MethodVarMethod.cs
namespace Domain.Indirect.AllowedFrom
{
	public class CallInstanceMethodIndirectIndirect_MethodVarMethod
	{
        private Domain.Indirect.Intermediate.BackgroundService bckg;

		private string test;

		public void Case6()
        {
            bckg = new Domain.Indirect.Intermediate.BackgroundService();
			test = (string)bckg.GetServiceTwo().serviceOne.GetName();
		}
	}
}
